src/sentry/static/sentry/app/views/projectDetailsLayout.jsx

```diff
@@ -1,21 +1,36 @@
 import React from 'react';
-
+import Reflux from 'reflux';
 import createReactClass from 'create-react-class';
 
+import EnvironmentStore from '../stores/environmentStore';
+import LatestContextStore from '../stores/latestContextStore';
+
 import DiffModal from '../components/modals/diffModal';
 import ProjectHeader from '../components/projectHeader';
 import ProjectState from '../mixins/projectState';
 
 const ProjectDetailsLayout = createReactClass({
   displayName: 'ProjectDetailsLayout',
-  mixins: [ProjectState],
+  mixins: [
+    ProjectState,
+    Reflux.connect(EnvironmentStore, 'environments'),
```

Comment:
I could be wrong but it looks we have this layout with lots of children that will get re-rendered when it seems that only the environment dropdown menu inside of ProjectHeader needs these values from the store. If this is the case then IMO this would be better if we had a new component for the dropdown and have it connect to the store.

Reply:
Hmm I guess we need the entire component tree to re-render in order to re-fetch data...

Reply:
Yep, though I don't think that's what's causing re-render since this prop isn't being passed to those children